src/software/backend/wifi_backend.cpp

```diff
@@ -27,6 +27,14 @@ WifiBackend::WifiBackend(std::shared_ptr<const NetworkConfig> network_config)
         DynamicParameters->getNetworkConfig()->NetworkInterface()->value();
     int channel = DynamicParameters->getNetworkConfig()->Channel()->value();
 
+    MutableDynamicParameters->getMutableNetworkConfig()
+        ->mutableChannel()
+        ->registerCallbackFunction([this](int new_channel) {
+            std::string new_network_interface =
+                DynamicParameters->getNetworkConfig()->NetworkInterface()->value();
+            joinMulticastChannel(new_channel, new_network_interface);
+        });
```

Comment:
you can use the immutable config to register callbacks as soon as #1543 goes in

Reply:
sick. Hopefully this will end up going in first, but much appreciated. Didn't realize it would be such an easy fix :fire: